elasticdl/python/master/servicer.py

```diff
@@ -402,13 +405,13 @@ def ReportGradient(self, request, _):
 
             # other grads
             for k, v in tmp.items():
-                if k in self._gradient_sum:
+                if not self._use_async and k in self._gradient_sum:
                     self._gradient_sum[k] = self._gradient_sum[k] + v
```

Comment:
We only need to check `self._use_async` once. Checking `self._use_async` inside the `for` loop seems that it will hurt the performance.